repl/src/main/scala/com/cloudera/livy/repl/ReplDriver.scala

```diff
@@ -83,4 +84,32 @@ class ReplDriver(conf: SparkConf, livyConf: RSCConf)
     return session.state.toString
   }
 
+  override def createWrapper(msg: BaseProtocol.BypassJobRequest): BypassJobWrapper = {
+    kind match {
+      case Spark() => super.createWrapper(msg)
+      case PySpark() | PySpark3() => {
+        new BypassJobWrapper(this, msg.id,
+          new BypassPySparkJob(msg.serializedJob, PythonInterpreter.getPySparkJobProcessor))
+      }
+    }
+  }
+
+  override def addFile(path: String): Unit = {
+    kind match {
+      case Spark() => super.addFile(path)
+      case PySpark() | PySpark3() => PythonInterpreter.getPySparkJobProcessor.addFile(path)
+    }
+  }
+
+  override def addJarOrPyFile(path: String): Unit = {
+    kind match {
+      case Spark() => super.addJarOrPyFile(path)
+      case PySpark() | PySpark3() => {
+        val localCopyDir = new File(PythonInterpreter.getPySparkJobProcessor.getLocalTmpDirPath)
+        doAddJarOrPyFile_copyFileToLocal(localCopyDir, path,
+          SparkContext.getOrCreate(conf))
+        PythonInterpreter.getPySparkJobProcessor.addPyFile(path)
```

Comment:
Have you tested this? I'm looking at the implementation of `addPyFile` and it's unclear to me that it would work if you're adding an HDFS path here. The goal is to have the new module be available in the driver, and my hunch is that it will only work if it's a local path (i.e. the return value of `doAddJarOrPyFile_copyFileToLocal`).